internal/pkg/runtime/launcher/oci/mounts_linux.go

```diff
@@ -625,13 +625,19 @@ func (l *Launcher) addBindMount(mounts *[]specs.Mount, b bind.Path, allowSUID bo
 		)
 	}
 
+	// /proc and /sys need nosuid,noexec,nodev set explicitly for runc
+	// See: https://github.com/opencontainers/runc/discussions/3801
 	opts := []string{"rbind", "nodev"}
+	if strings.HasPrefix(b.Source, "/proc") || strings.HasPrefix(b.Source, "/sys") {
+		opts = append(opts, "nosuid", "noexec")
+	} else {
+		if !allowSUID {
+			opts = append(opts, "nosuid")
```

Comment:
This is low priority, but: I don't love the multiple paths of boolean "descent" that can lead to `nosuid`. I think I'd prefer if there was a set(/map) of options, so that adding something like `nosuid` twice would have no ill effects, and then when it's time to build the actual options string, the members of that set are joined with commas. It'd simplify this code and, esp., reading this code, imo.

Reply:
I see your point. I think if we are going to simplify in this manner then I'd prefer to have our own struct, wrapping specs.Mount, that provides something like an `.addOptions(...)` function.

If you'd like to take that on that'd be good.

What I don't want to do is add temporary maps, and map <=> slice conversion code inside multiple existing functions.